charley_tran/week_01/day04_hw.html

```diff
@@ -0,0 +1,152 @@
+<html>
+<script>
+/*Part 1, Rectangle
+
+Given the following a rectangle object like the one below, write the following functions:
+
+isSquare - Returns whether the rectangle is a square or not
+area - Returns the area of the rectangle
+perimeter - Returns the perimeter of the rectangle
+var rectangle = {
+  length: 4,
+  width: 4
+};*/
+
+var rectangle = {
+	length : 4,
+	width : 4
+};
+
+var isSquare = function() {
+	if (rectangle.length = rectangle.width) {
+		return rectangle = 'square'
+	}
+}
+var area = function() {
+	return rectangle.length * rectangle.width
+}
+
+var perimeter = function () {
+	return 2 * (rectangle.length + rectangle.width)
+}
+
+//part 2 , triangle.
+
+var triangle = {
+  sideA: 3,
+  sideB: 4,
+  sideC: 4
+};
+
+var isEquilateral = function() {
+	if (triangle.sideA == triangle.sideB && triangle. sideA === triangle.sideC) {
+		return console.log('triangle is equilateral')
+	} else return console.log('triangle is not equilateral.')
+}
+
+var isIsosceles = function() {
+	if (triangle.sideA == triangle.sideB && triangle. sideA === triangle.sideC) {
+		return console.log('triangle is isosceles')
+	} else if (triangle.sideA == trianglesideB || triangle.sideB == triangle.sideC || triangle.sideC == triangle.sideA) {
+		return console.log('triangle is isosceles')
+	} else return console.log('triangle is not iscosceles')
+}
+
+var areaTriangle = function() {
+	var S = triangle.sideA + triangle.sideB + triangle.sideC;
+	return Math.sqrt(S*(S-triangle.sideA)*(S-triangle.sideB)*(S-triangle.sideC))
+}
+
+
+
+var isObtuse = function() {
+	if (triangle.sideA !== triangle.sideB && triangle.sideB !== triangle.sideC) {
+		return console.log ('triangle is obtuse')
+	} else {
+		console.log('triangle is not obtuse')
+	}
+}
+
+
+// Bank homework
+// 1 bank. bank as accounts
+// var bank = {
+// 	account 1 : 0
+// 	account 2 : 0
+// }
+
+var bank = {
+	test : 0,
+	charley : 100
+};
+
+
+for (var key in bank) {
+  console.log("obj." + key + " = " + bank[key]);
+}
+
+var bankBalance = function () {
+for (var key in bank) {
+	var a = 0;
```

Comment:
you should assign var a = 0; outside of the for loop, otherwise it becomes 0 on each iteration. This is why accountBalance will always equal to last value from the bank object. PS: try it adding few more people as a key and some sums as the values and you ll see that each time your accountBalance will show only last value.